aiida/calculations/transfer.py

```diff
@@ -0,0 +1,250 @@
+    instance of this CalcJob (so you can't directly move files from remote to remote and all
+    RemoteData nodes must be from the same machine).
+
+    The user needs to provide three inputs:
+
+        * ``instructions``: a dict node specifying which files to copy from which nodes.
+        * ``source_nodes``: a dict of nodes, each with a unique identifier label as its key.
+        * ``metadata.computer``: the computer that contains the remote files and will contain
+          the final RemoteData node.
+
+    The ``instructions`` dict must have the ``retrieve_files`` flag. The CalcJob will create a
+    new folder in the remote machine (RemoteData) and put all the files there and will either:
+
+        (1) leave them there (``retrieve_files = False``) or ...
+        (2) create a local FolderData and copy there all the files (``retrieve_files = True``)
+
+    The `instructions` dict must also contain at least one list with specifications of which files
+    to copy and from where. All these lists take tuples of 3 that have the following format:
+
+    .. code-block:: python
+
+        [ ..., ( source_node_key, path_to_file_in_source, path_to_file_in_target), ... ]
+
+    where the ``source_node_key`` has to be the respective one used when providing the node in the
+    ``source_nodes`` input nodes dictionary.
+
+
+    The two main lists to include are ``local_files`` (for files to be taken from FolderData nodes)
+    and ``remote_files`` (for files to be taken from RemoteData nodes). Alternatively, files inside
+    of RemoteData nodes can instead be put in the ``symlink_files`` list: the only difference is that
+    files from the first list will be fully copied in the target RemoteData folder, whereas for the
+    files in second list only a symlink to the original file will be created there. This will only
+    affect the content of the final RemoteData target folder, but in both cases the full file will
+    be copied back in the local target FolderData (if ``retrieve_files = True``).
+    """
+
+    @classmethod
+    def define(cls, spec):
+        super().define(spec)
+
+        spec.input(
+            'instructions',
+            valid_type=orm.Dict,
+            help='A dictionary containing the `retrieve_files` flag and at least one of the file lists:'
+            '`local_files`, `remote_files` and/or `symlink_files`.',
+            validator=validate_instructions,
+        )
+        spec.input_namespace(
+            'source_nodes',
+            valid_type=(orm.FolderData, orm.RemoteData),
+            dynamic=True,
+            help='All the nodes that contain files referenced in the instructions.',
+        )
+
+        # The transfer just needs a computer, the code are resources are set here
+        spec.inputs['code'].required = False
+        spec.inputs['metadata']['computer'].required = True
+        spec.inputs['metadata']['options']['resources'].default = {
```

Comment:
Same here, since we are not going to use `resources` can we just pop this port so we also don't have to specify the default?

Reply:
I guess the same argument as [the one for code](https://github.com/aiidateam/aiida-core/pull/4194#discussion_r527602924) would apply here. Although now that we are skipping the submit/update, I could instead of setting a default do `required = False`.

Reply:
Mmm, so apparently the `scheduler.preprocess_resources` is being called by the `presubmit` routine, so it is still raising errors here...I guess I could try to disable this when `skip_submit` is true, let me try.

Reply:
So, from what I could see, the problem is that the presubmit step does a lot of scheduler setups which are intercalated with other things like copy_list checks. It is hard to say on a first approximation which ones can be taken out by an if clause without impacting on the rest of the code. In principle, comenting [these lines](https://github.com/aiidateam/aiida-core/blob/f945dcabbf82a921e9b844b1865681e1a945ef67/aiida/engine/processes/calcjobs/calcjob.py#L506-L516) seems to work:

```python
        if not calc_info.skip_submit:
            resources = self.node.get_option('resources')
            scheduler.preprocess_resources(resources, computer.get_default_mpiprocs_per_machine())
            job_tmpl.job_resource = scheduler.create_job_resource(**resources)

            subst_dict = {'tot_num_mpiprocs': job_tmpl.job_resource.get_tot_num_mpiprocs()}

            for key, value in job_tmpl.job_resource.items():
                subst_dict[key] = value
            mpi_args = [arg.format(**subst_dict) for arg in computer.get_mpirun_command()]
            extra_mpirun_params = self.node.get_option('mpirun_extra_params')  # same for all codes in the same calc
```

However, I feel like just if-ing like this would unnecessarily add complexity "gunk" to the code and would just complicate further modifications of the section. On the other hand, doing a full reorganization of this routine would involve a substantial ammount of work which I'm not sure is fully related to the objective of this PR. I can try to do this if you consider this critical.

Reply:
Fair point, I agree then that this is not the place to do it, so you can consider this comment resolved and leave the code here as is